File: Problems/LeetCode - Two sum/SolutionTest.java
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SolutionTest {
    @Test
    public void testNullArray() {
        // perform
        int[] result = createSolution().twoSum(null, 0);

        // check
        assertEquals(0, result.length);
    }

    @Test
    public void testEmptyArray() {
        // perform
        int[] result = createSolution().twoSum(new int[0], 0);

        // check
        assertEquals(0, result.length);
    }

    @Test
    public void testArrayWithOneElement() {
        // perform
        int[] result = createSolution().twoSum(new int[1], 0);

        // check
        assertEquals(0, result.length);
    }

    @Test
    public void testArrayNumber1() {
        // perform
        int[] result = createSolution().twoSum(new int[]{2, 7, 11, 15}, 9);

        // check
        assertEquals(0, result[0]);
        assertEquals(1, result[1]);
    }

    private Solution createSolution() {
        return new Solution();
    }
}

File: Problems/LeetCode - PhoneLetterCombinations/SolutionTest.kt
import org.junit.Test
import org.junit.Assert.*

class SolutionTest {
    @Test
    fun letterCombinations() {
        // prepare
        val solution = createSolution()

        // perform
        val result = solution.letterCombinations("23")

        // check
        val resultShouldBe = listOf("ad", "ae", "af", "bd", "be", "bf", "cd", "ce", "cf")
        assertEquals(true, resultShouldBe.toSortedSet() == result.toSortedSet())
    }

    private fun createSolution() = Solution()
}